### PR TITLE
ci: Bump the macOS image to the latest Monterey

### PR DESCRIPTION
**Pull request by hebasto**
We usually follow the latest available macOS image.

### CODE REVIEW

**Review by Zero-1729 — COMMENTED** [review]
Light crACK 34fd25eaf0b7cb1e12bdc0f4c3f4a95e32a3434c

File: .cirrus.yml [hunk 1]
```diff
@@ -282,13 +282,13 @@ task:
     FILE_ENV: "./ci/test/00_setup_env_mac.sh"
 
 task:
-  name: 'macOS 11 native [gui, sqlite only] [no depends]'
+  name: 'macOS 12 native [gui, system sqlite only] [no depends]'
   brew_install_script:
-    - brew install boost libevent qt@5 miniupnpc libnatpmp ccache zeromq qrencode sqlite libtool automake pkg-config gnu-getopt
+    - brew install boost libevent qt@5 miniupnpc libnatpmp ccache zeromq qrencode libtool automake gnu-getopt
```
Comment (MarcoFalke) [inline .cirrus.yml:287]: Why is this change needed and what does it effect?

Comment (hebasto) [reply .cirrus.yml:287]: Homebrew's `pkg-config` has already been install in this image.

https://github.com/bitcoin/bitcoin/blob/master/doc/build-osx.md#wallet-dependencies:
> Apple has included a useable `sqlite` package since macOS 10.14. You may not need to install this package.
